### ISSUE TITLE
Loss going down with training but precision and recall stay at zero

### ISSUE THREAD

**Issue by victoryc** [facebookresearch/votenet#45]
I tried training Votenet for some outdoor scene data we have. The weird thing I am noticing is that though the loss has gone down considerably with nearly a day of training, the precision and recall numbers are consistently staying at zero.

Any ideas/insights?

eval mean box_loss: 1.709604
eval mean center_loss: 1.709604
eval mean heading_cls_loss: 0.000000
eval mean heading_reg_loss: 0.000000
eval mean loss: 104.791534
eval mean neg_ratio: 0.990234
eval mean obj_acc: 1.000000
eval mean objectness_loss: 0.000164
eval mean pos_ratio: 0.000000
eval mean sem_cls_loss: 0.000000
eval mean size_cls_loss: 0.000000
eval mean size_reg_loss: 0.000000
eval mean vote_loss: 8.769468
0 0
2 0
4 0
1 0
5 0
3 0
eval Car Average Precision: 0.000000
eval Tree Average Precision: 0.000000
eval Building Average Precision: 0.000000
eval mAP: 0.000000
eval Car Recall: 0.000000
eval Tree Recall: 0.000000
eval Building Recall: 0.000000

Weirdly enough, I got these zero values for precision and recall, even when I used a couple of scenes from the training data itself for evaluation. At least for those, I would have expected the recall & precision to be good. Of course, eventually, the goal is to evaluate it on data that hasn't been used for training. This was just a test to see whether it gives non-zero precision/recall values at least for scenes used in training.

**Comment by charlesq34** [comment]
Hi @victoryc 

It seems in your data, the pos_ratio is 0, which means all surface points are pretty far from the votes [see link].(https://github.com/facebookresearch/votenet/blob/257b8d8369511c6b4e0cea9c33ef0da01dd10f77/models/loss_helper.py#L222)

You need to check if the voting loss decreases normally (if voting is poor, then there will be no box regression supervision).

You can also train a box net to see if you get anything there.

**Comment by JerryXu0907** [comment]
@victoryc Hi, did you manage to figure out what's going wrong?

**Comment by nazmicancalik** [comment]
Hey @victoryc, did you manage to find a solution to your problem? I would also be interested in it :)
Best

**Comment by Mif-cl** [comment]
@victoryc Hi, could you update any solutions?
